File: March-02.py
class Solution:
    def firstElementKTime(self, n, k, a):
        fdh = {}
        for i in range(n):
            if fdh.get(a[i]) and fdh[a[i]] + 1 == k:
                return a[i]
            elif fdh.get(a[i]):
                fdh[a[i]] += 1
            else:
                fdh[a[i]] = 1
        return -1

File: March 2024 GFG SOLUTION/March-02.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

class Solution{
    public:
    int firstElementKTime(int n, int k, int a[])
    {
        std::unordered_map<int, int> map;
        for(int i=0; i<n; i++){
            map[a[i]]++;
            if(map[a[i]] == k)
                return a[i];
        }
        return -1;
    }
};

//{ Driver Code Starts.

int main() {
	int t;
	cin >> t;
	while(t--){
	    int n, k;
	    cin >> n >> k;
	    int a[n];
	    for(int i = 0;i<n;i++){
	        cin >> a[i];
	    }
	    Solution ob;
	    cout<<ob.firstElementKTime(n, k, a)<<endl;
	}
	
	return 0;
}
// } Driver Code Ends
